src/content/advanced/tls-certificates/index.md

```diff
@@ -0,0 +1,104 @@
+weight: 20
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-cabbage
+user_questions:
+  - How do I install cert-manager?
+  - How do I obtain TLS certificates?
+last_review_date: 2022-07-19
+---
+
+Exposing HTTP services using an Ingress is a pretty straightforward task in Kubernetes. Using plain HTTP is, however, discouraged in favor of using HTTPS. `cert-manager` provides a simple, declarative way to automatically obtain TLS certificates for the endpoints we expose.
+
+Our App Platform provides a `cert-manager-app` that users can install. In order to configure it for TLS certificate management, we should install a `ClusterIssuer` that solves DNS challenges. Note that this also works for internal Load Balancers.
+
+We can do so by, first of all, disabling the default `ClusterIssuer` when installing the App, by providing the following user values:
+
+```yaml
+global:
+  giantSwarmClusterIssuer:
+    install: false
+```
+
+Then, we can create a new Cluster Issuer that solves DNS challenges. For instance, on Azure:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-giantswarm
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: test@customer.com
+    privateKeySecretRef:
+      name: letsencrypt-giantswarm
+    solvers:
+    - dns01:
+        azureDNS:
+          # find the right values in `azure.yaml` in the `external-dns` deployment
+          subscriptionID: subscription-id
+          resourceGroupName: cluster-id
+          hostedZoneName: cluster-id.k8s.installation.region.provider.gigantic.io
+```
+
+Finally, we can simply create an Ingress. It is important to specify the correct `spec.ingressClassName` (which will likely be either `nginx` or `nginx-internal`) and `spec.tls` fields.
+
+An example Ingress would look like this:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: test
+  annotations:
+    kubernetes.io/tls-acme: "true"
+spec:
+  ingressClassName: nginx
+  tls:
+  - hosts:
+    - test.cluster.k8s.installation.region.provider.gigantic.io
+    secretName: test-tls
+  rules:
+  - host: test.cluster.k8s.installation.region.provider.gigantic.io
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: hello-world-service
+            port:
+              number: 8080
+```
+
+It is also possible to obtain wildcard certificates, which can be useful in some cases:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: wildcard
+spec:
+  commonName: "*.cluster.k8s.installation.region.provider.gigantic.io"
+  dnsNames:
+  - "*.cluster.k8s.installation.region.provider.gigantic.io"
+  issuerRef:
+    group: cert-manager.io
+    kind: ClusterIssuer
+    name: letsencrypt-giantswarm
+  secretName: wildcard-tls
+```
+
+this certificate can then be used by setting the following `spec.tls` field in an Ingress:
```

Comment:
I think this shouldn't be separate.

Usually setting `cert-manager.io/cluster-issuer: nameOfClusterIssuer` annotation on `Ingress` resources makes cert-manager create a `Certificate`

Reply:
mh oh ok, I see this is about the wildcard certificate. Unsure what to do here. Maybe thats okay then to leave it like this

Reply:
You probably don't want to issue a wildcard certificate from an ingress because both the `spec.tls.hosts` and `spec.rules.host` must match. So creating a wildcard certificate from you ingress requires this ingress to also match all hosts. But in some cases you just want one wildcard certificate for your (sub) domain and multiple single host name ingresses sharing the same wildcard certificate.

This way `cert-manager` doesn't even need to care about ingress resources and the `cert-manager` annotations can be omitted.

Reply:
Could you also point out that the annotation should be omitted when the certificate has been created "manually"?

Reply:
Which annotation? `kubernetes.io/tls-acme: "true"`?